app/controllers/administrative-units/administrative-unit/sites/site/edit.js

```diff
@@ -0,0 +1,76 @@
+import Controller from '@ember/controller';
+import { dropTask } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class AdministrativeUnitsAdministrativeUnitSitesSiteEditController extends Controller {
+  @service router;
+  @tracked isPrimarySite =
+    this.site.get('id') === this.administrativeUnit.get('primarySite.id');
+
+  get site() {
+    return this.model.adminUnitSite.site;
+  }
+
+  get administrativeUnit() {
+    return this.model.adminUnitSite.administrativeUnit;
+  }
+
+  @dropTask
+  *save(event) {
+    event.preventDefault();
+
+    let address = yield this.site.address;
+
+    if (address.hasDirtyAttributes) {
+      let fullStreet = `${address.street || ''} ${address.number || ''} ${
+        address.boxNumber || ''
+      }`.trim();
+
+      let muncipalityInformation = `${address.postcode || ''} ${
+        address.municipality || ''
+      }`.trim();
+
+      if (fullStreet && muncipalityInformation) {
+        address.fullAddress = `${fullStreet}, ${muncipalityInformation}`;
+      } else if (fullStreet) {
+        address.fullAddress = fullStreet;
+      } else {
+        address.fullAddress = muncipalityInformation;
+      }
+
+      yield address.save();
+    }
+
+    let contacts = yield this.site.contacts;
+    if (contacts.firstObject.hasDirtyAttributes) {
+      yield contacts.firstObject.save();
+    }
+
+    yield this.site.save();
+
+    if (!this.isPrimarySite) {
+      this.administrativeUnit.sites.pushObject(this.site);
```

Comment:
I still think it makes more sense to store _all_ sites in the sites relationship, so also the one that is marked the primary-site. That way we only need to add or remove the primary site relationship without touching the `sites` one. It just seems weird to not get the primary site when accessing the sites relationship.

Reply:
I need to change the data, then.

Reply:
Yea, maybe we should discuss that with other people first. I didn't define the data model so my expectations might not match the reallity :smile:.